srcpkgs/sile/template

```diff
@@ -1,27 +1,39 @@
 # Template file for 'sile'
 pkgname=sile
-version=0.9.5.1
-revision=9
+version=0.15.5
+revision=1
 build_style=gnu-configure
-hostmakedepends="pkg-config"
+build_helper=rust
+configure_args="--with-system-lua-sources --with-system-luarocks"
+hostmakedepends="jq poppler cargo rust pkg-config"
 makedepends="harfbuzz-devel lua51-devel lua51-lpeg lua51-luaexpat
- lua51-zlib lua51-luafilesystem lua51-luasocket lua51-luasec"
-depends="lua51-lpeg lua51-luaexpat lua51-zlib lua51-luafilesystem
- lua51-luasocket lua51-luasec"
+ lua51-zlib lua51-luafilesystem lua51-luasocket lua51-luasec lua51-cassowary
+ lua51-compat53 lua51-cldr lua51-fluent lua51-loadkit lua51-linenoise
+ lua51-cliargs lua51-luarepl lua51-luasocket lua51-luasec lua51-luautf8
+ lua51-vstruct fontconfig-devel LuaJIT-devel font-sil-gentium rust-std"
+depends="LuaJIT lua51-lpeg lua51-luaexpat lua51-zlib lua51-luafilesystem lua51-cassowary
+ lua51-luasocket lua51-luasec lua51-compat53 lua51-cldr lua51-fluent lua51-loadkit
+ lua51-linenoise lua51-cliargs lua51-luarepl lua51-luasocket lua51-luasec lua51-luautf8
+ lua51-vstruct font-sil-gentium"
 short_desc="Modern typesetting system inspired by TeX"
 maintainer="John <me@johnnynator.dev>"
 license="MIT"
 homepage="http://www.sile-typesetter.org/"
-distfiles="https://github.com/simoncozens/sile/releases/download/v${version}/sile-${version}.tar.bz2"
-checksum=60cdcc4509971973feab352dfc1a86217cc1fdb12d56823f04d863afef92003a
+distfiles="https://github.com/simoncozens/sile/releases/download/v${version}/sile-${version}.tar.zst"
+checksum=d20137b02d16302d287670fd285ad28ac3b8d3af916460aa6bc8cbff9321b9f9
 
 if [ "$CROSS_BUILD" ]; then
-	hostmakedepends+=" lua51 lua51-devel lua51-lpeg luaexpat lua51-zlib luafilesystem
-	 luasocket lua51-luasec"
+	hostmakedepends+=" ${makedepends}"
 fi
 
 post_install() {
-	vlicense LICENSE
+	vlicense LICENSE.md
+}
+
+post_configure() {
+	if [ "$CROSS_BUILD" ]; then
+		vsed -i '1766s|$<|target/*linux*/release/$@|' Makefile.in
```

Comment:
Is this something we need to fix upstream to make cross compiling work better out of the box?

Reply:
I am not familiar with automake.
and how this line is being generated.
with current approach, i have to rematch line number for cross building

Reply:
I see what's going on. We're assuming Cargo output in `target/release` but when cross compiling the target host gets added in there, `target/x86_64-unknown-linux-gnu/release`. We have a way to override the target dir that you could probably overload to address this just a touch easier:

```console
$ sed -i '/^RUST_TARGET_SUBDIR/s#= #= x86_64-unknown-linux-gnu/#' Makefile.in
```

Note you would have to specify the full host identifier, not use a wildcard like in your current hack.

This is just stuffing the host info in as a path segment before the "target" part. It's still a hack, and as far as I see we don't have a way to catch that right now. I'll look into it from our side.

Reply:
https://github.com/sile-typesetter/sile/issues/2099

Reply:
```
$ sed -i '/^RUST_TARGET_SUBDIR/s#= #= x86_64-unknown-linux-gnu/#' Makefile.in
```

this did not worked as expected

Reply:
Yes there seem to be several possible scenarios and Cargo doesn't make it easy to figure out which one is at play. Cargo never has played nice with other build systems. We have an upstream issue open for tracking a proper fix to this, so I think you can just hang on to your workaround until we get it fixed in an upstream release.